api/envoy/extensions/http/header_validators/envoy_default/v3/header_validator.proto

```diff
@@ -115,4 +135,8 @@ message HeaderValidatorConfig {
   // Envoy will respond with 400 to requests with disallowed methods.
   // By default methods with arbitrary names are accepted.
   bool restrict_http_methods = 3;
+
+  // Action to take when a client request with a header name containing underscore characters is received.
+  // If this setting is not specified, the value defaults to ALLOW.
+  HeadersWithUnderscoresAction headers_with_underscores_action = 4;
```

Comment:
High-level question: do you think there may be a need in the future for action-specific-settings.
For example, will there be something like "REPLACE_UNDERSCORE" that will need the replacement string.
If so, I would suggest using a oneof, and messages for each option.

Reply:
All these settings are for existing behavior. In the future if people want to modify it, they can add a new extension with their own config.